x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -568,16 +567,17 @@ private SingleForecast forecast(Metadata metadata, IndexAbstraction.DataStream s
             DataStream dataStream = stream.getDataStream();
             for (int i = 0; i < numberNewIndices; ++i) {
                 final String uuid = UUIDs.randomBase64UUID();
-                final Tuple<String, Long> dummyRolledDatastream = dataStream.nextWriteIndexAndGeneration(state.metadata());
```

Comment:
Maybe add a `unsafeNextWriteIndexAndGeneration(...)` and use it here?
That way the exact same logic remains to be used.

Reply:
Sure, see da77aaee1931f03c82ddfcd50ec93fc86bd00d9d